emmet-core/emmet/core/electrode.py

```diff
@@ -0,0 +1,136 @@
+""" Core definition of an Electrode Document """
+from typing import List, Dict, ClassVar
+from datetime import datetime
+
+from pydantic import BaseModel, Field
+
+from emmet.stubs import ComputedEntry
+from emmet.core.structure import StructureMetadata
+
+
+class ElementEvolution(BaseModel):
+    """
+    Result of the PhaseDiagram.get element_profile which moves a along a line in
+    composition space and return the reaction, chemical potential, and the remaining
+    amount of the element
+    """
+
+    reaction: str = Field(
+        ...,
+        description="Evolution reaction determined by a facet of the phase diagram",
+    )
+
+    chempot: float = Field(
+        ...,
+        description="The chemical potential of the target element solved on that facet",
+    )
+
+    evolution: float = Field(
+        ..., description="Remaining amount of element .. TODO get better description",
+    )
+
+
+class VoltagePair(BaseModel):
+    """
+    Based on AbstractVoltagePair from pymatgen
+    """
+
+    voltage: float = Field(
+        ..., description="Voltage of voltage pair",
+    )
+
+    mAh: float = Field(
+        ..., description="Energy in mAh",
+    )
+
+    mass_charge: float = Field(
+        ..., description="Mass of charged pair.",
+    )
+
+    mass_discharge: float = Field(
+        ..., description="Mass of discharged pair.",
+    )
+
+    vol_charge: float = Field(
+        ..., description="Volume of charged pair.",
+    )
+
+    vol_discharge: float = Field(
+        ..., description="Volume of discharged pair.",
+    )
+
+    frac_charge: float = Field(
+        ..., description="Atomic Fraction of working ion in charged pair.",
+    )
+
+    frac_discharge: float = Field(
+        ..., description="Atomic Fraction of working ion in discharged pair.",
+    )
+
+    working_ion_entry: float = Field(
+        ..., description="Working ion as an entry.",
+    )
+
+
+class Electrode(BaseModel):
+    """
+    Based on AbstractElectrode from pymatgen
+    """
+
+    battery_id: str = Field(..., description="The id for this battery document.")
+
+    framework: StructureMetadata = Field(
```

Comment:
`StructureMetadata` doesn't contain a structure, so you can either have `Electrode` inherit from `StructureMetadata` and choose the charged or discharged end as the reference structure; or you can store an explicit `framework_metdata` key that is `StructureMetadata` and `framework` is of type `Structure`